e2e/compatibility/terraform_download_test.go

```diff
@@ -0,0 +1,65 @@
+//go:build e2e
+// +build e2e
+
+package compatibility
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/hashicorp/consul-terraform-sync/driver"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// This technically isn't an e2e test, but it downloads a file from the internet,
+// so it likely shouldn't be run frequently with other tests.
+func TestCompatibility_TFDownload(t *testing.T) {
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "Compatibility_TFDownload")
+	cleanup := testutils.MakeTempDir(t, tempDir)
+	defer cleanup()
+
+	conf := config.DefaultTerraformConfig()
+	conf.Path = &tempDir
+	// Use blank as a default value. Nil results in a panic.
+	tfVersion := ""
+	conf.Version = &tfVersion
+
+	binaryPath := path.Join(tempDir, "terraform")
+	info, err := os.Stat(binaryPath)
+	// We want to ensure the file does not exist and remove it if found.
+	if !os.IsNotExist(err) {
+		require.NoError(t, err, "An unexpected error occurred while checking for old terraform executable.")
+	}
+	if info != nil {
```

Comment:
Spacing out the two if blocks so the second one doesn't get confused with an `else` 
```suggestion

	if info != nil {
```